crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2880,64 +2888,298 @@ impl TransactionValidator for Backend {
-        transaction.nonce = info.as_ref().unwrap().nonce;
-    }
+) -> AnyRpcTransaction {
+    if let TypedTransaction::Deposit(ref deposit_tx) = eth_transaction.transaction {
+        let DepositTransaction {
+            nonce: _,
+            source_hash,
+            from,
+            kind,
+            mint,
+            gas_limit,
+            is_system_tx,
+            input,
+            value,
+        } = deposit_tx.clone();
+
+        let dep_tx = TxDeposit {
+            source_hash,
+            input,
+            from,
+            mint: Some(mint.to()),
+            to: kind,
+            is_system_transaction: is_system_tx,
+            value,
+            gas_limit,
+        };
 
-    if eth_transaction.is_dynamic_fee() {
-        if block.is_none() && info.is_none() {
-            // transaction is not mined yet, gas price is considered just `max_fee_per_gas`
-            transaction.gas_price = transaction.max_fee_per_gas;
-        } else {
-            // if transaction is already mined, gas price is considered base fee + priority fee: the
-            // effective gas price.
-            let base_fee = base_fee.map_or(0u128, |g| g as u128);
-            let max_priority_fee_per_gas = transaction.max_priority_fee_per_gas.unwrap_or(0);
-            transaction.gas_price = Some(base_fee.saturating_add(max_priority_fee_per_gas));
+        let ser = serde_json::to_value(&dep_tx).unwrap();
+        let mut fields = OtherFields::default();
+
+        if let serde_json::Value::Object(map) = ser {
+            for (k, v) in map {
+                fields.insert(k, v);
+            }
+
+            let inner = UnknownTypedTransaction {
+                ty: AnyTxType(DEPOSIT_TX_TYPE_ID),
+                fields,
+                memo: Default::default(),
+            };
+
+            let envelope =
+                AnyTxEnvelope::Unknown(UnknownTxEnvelope { hash: eth_transaction.hash(), inner });
+
+            let tx = Transaction {
+                inner: envelope,
+                block_hash: block
+                    .as_ref()
+                    .map(|block| B256::from(keccak256(alloy_rlp::encode(&block.header)))),
+                block_number: block.as_ref().map(|block| block.header.number),
+                transaction_index: info.as_ref().map(|info| info.transaction_index),
+                effective_gas_price: None,
+                from,
+            };
+
+            return WithOtherFields::new(tx);
         }
-    } else {
-        transaction.max_fee_per_gas = None;
-        transaction.max_priority_fee_per_gas = None;
     }
 
-    transaction.block_hash =
-        block.as_ref().map(|block| B256::from(keccak256(alloy_rlp::encode(&block.header))));
+    let transaction: Transaction = eth_transaction.clone().into();
+
+    let envelope = transaction.inner;
 
-    transaction.block_number = block.as_ref().map(|block| block.header.number);
+    let envelope = match envelope {
```

Comment:
this match should only do `hash` altering. `nonce` and `to` changes doesn't make much sense and the `gas_price` should be moved to `effective_gas_price`

Reply:
Addressed in [37a97b4](https://github.com/foundry-rs/foundry/pull/9280/commits/37a97b4051876960ce1c32bdb24703a875126071)